tommath_private.h

```diff
@@ -10,6 +10,42 @@
 extern "C" {
 #endif
 
+/* Hardening libtommath
+ * --------------------
+ *
+ * By default memory is zeroed before calling
+ * MP_FREE to avoid leaking data. This is good
+ * practice in cryptographical applications.
+ *
+ * Note however that memory allocators used
+ * in cryptographical applications can often
+ * be configured by itself to clear memory,
+ * rendering the clearing in tommath unnecessary.
+ * See for example https://github.com/GrapheneOS/hardened_malloc
+ * and the option CONFIG_ZERO_ON_FREE.
+ *
+ * Furthermore there are applications which
+ * value performance more and want this
+ * feature to be disabled. For such applications
+ * define MP_NO_ZERO_ON_FREE during compilation.
+ */
+#ifdef MP_NO_ZERO_ON_FREE
+#  define MP_FREE_BUFFER(mem, size)   MP_FREE((mem), (size))
```

Comment:
Second argument to `MP_FREE` is `size_t`. Shouldn't it get at least a cast?

Reply:
I'd say it shouldn't get a cast here, but where required in the sources where the invocation of the macro happens.
Otherwise it's possible that something like the following complies without even throwing a warning
```
void* foo = malloc(123);
MP_FREE_BUFFER(foo, foo);
```

Reply:
Wanted to restrict my otherwise quite verbose comments, and now I'm too shrt.

The last build failed because of a missing cast in `MP_FREE` and the culprits in the code causing it are still missing their rations of plaster-of-paris (e.g.: in `mp_fwrite()`).
Something has been done in the default branch here regarding that problem, so I asked myself why it is in one branch only and then decided to share that question publicly.